editor/src/components/inspector/controls/slider-control.tsx

```diff
@@ -20,7 +20,101 @@ export type SliderControlProps = DEPRECATEDControlProps<number> & {
   DEPRECATED_controlOptions: DEPRECATEDSliderControlOptions
 }
 
-export class SliderControl extends React.Component<SliderControlProps> {
+export const SliderControl: React.FunctionComponent<SliderControlProps> = (props) => {
+  const [isSliding, setIsSliding] = React.useState(false)
+
+  const handleBeforeChange = React.useCallback(() => {
+    setIsSliding(true)
+  }, [])
+
+  const onChangeFn = props.onForcedSubmitValue ?? props.onSubmitValue
+
+  const handleOnAfterChange = React.useCallback(
+    (n: number) => {
+      onChangeFn(n)
+      setIsSliding(false)
+    },
+    [onChangeFn],
+  )
+
+  const controlOptions = {
+    ...defaultSliderControlOptions,
+    ...props.DEPRECATED_controlOptions,
+  }
+
+  let marks: Marks = {}
+  if (typeof controlOptions.origin === 'number') {
+    marks[controlOptions.origin] = {
+      style: {
+        display: 'none',
+      },
+      label: String(controlOptions.origin),
+    }
+  }
+  return (
+    <FlexRow
+      style={{
+        width: '100%',
+        height: UtopiaTheme.layout.inputHeight.default,
+        paddingLeft: 4,
+        paddingRight: 4,
+        alignItems: 'center',
+        ...props.style,
+      }}
+      className={props.controlClassName}
+      onContextMenu={props.onContextMenu}
+    >
+      <Slider
+        disabled={!props.controlStyles.interactive}
+        value={props.value}
+        onBeforeChange={handleBeforeChange}
+        onChange={props.onTransientSubmitValue}
+        // onAfterChange={props.onForcedSubmitValue ?? props.onSubmitValue}
+        onAfterChange={handleOnAfterChange}
+        min={controlOptions.minimum}
+        max={controlOptions.maximum}
+        step={controlOptions.stepSize}
+        marks={marks}
+        handleStyle={{
```

Comment:
This will also be a newly created object, which the Slider may re-render each time.

Reply:
Thanks! I've moved all the conditional styles in to useMemo fns - do I need to do that for the unconditional styles as well? Eg
```
dotStyle={{
          backgroundColor: props.controlStyles.railColor,
        }}
```

Reply:
I believe so, that's right isn't it @balazsbajorics ?

Reply:
for totally static styled, the easiest is to just put them in the file root. then they will only be generated once, when loading the editor.

btw I don't know if the `<Slider />` component is memoized. If it's not memoized, this is all moot

Reply:
These still depend on the props. I've made the changes, no idea about the internals of rc-slider but it seems our own Slider is not memoized. 

Btw is it possible that making these changes (useMemo) create a significant performance hit?